src/components/SearchCard.js

```diff
@@ -0,0 +1,200 @@
+  const [error, setError] = useState("");
+  const [defs, setDefs] = useState([]);
+  const [networkValue, setNetworkValue] = useState(defaultNetwork.value);
+
+  const resetData = () => {
+    setLoading(false);
+    setSuccess(false);
+    setInstalling(false);
+    setContract("");
+    setDefs([]);
+  };
+
+  const getNetwork = () =>
+    constants.CONTRACT_NETWORKS.find(({ value }) => value === networkValue) ??
+    defaultNetwork;
+
+  function fetchContractData(input) {
+    if (
+      input.slice(0, 2) !== "0x" ||
+      false === /^[0-9a-fA-F]+$/.test(input.slice(2)) ||
+      input.length !== 42
+    ) {
+      // Not a valid address
+      setError("Invalid Ethereum contract address");
+      resetData();
+    } else {
+      const networkToFetch = getNetwork();
+      fetch(`${networkToFetch.api}${input}`)
+        .then((response) => response.json())
+        .then((resp) => {
+          // Map confusing error strings to better descriptions
+          if (resp.result === "Contract source code not verified") {
+            resp.result = `Contract source code not published to ${networkToFetch.label} or not verified. Cannot determine data.`;
+          }
+          if (resp.status === "0") {
+            setError(resp.result);
+            resetData();
+          } else {
+            try {
+              const result = JSON.parse(resp.result);
+              const defs = session.client.parseAbi("etherscan", result, true);
+              setDefs(defs);
+              setContract(input);
+              setError("");
+              setSuccess(false);
+              setLoading(false);
+            } catch (err) {
+              setError(err.toString());
+              resetData();
+            }
+          }
+        })
+        .catch((err) => {
+          setError(err.toString());
+          resetData();
+        });
+    }
+  }
+
+  const throttledFetch = useMemo(
+    () => throttle(fetchContractData, 5100),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [networkValue]
+  );
+
+  function addDefs() {
+    setInstalling(true);
+    setError("");
+    // Longer timeout for loading these since requests may get dropped
+    session.client.timeout = 2 * constants.ASYNC_SDK_TIMEOUT;
+    session.addAbiDefs(defs, (err) => {
+      // Reset timeout to default
+      session.client.timeout = constants.ASYNC_SDK_TIMEOUT;
+      if (err) {
+        setError(err.toString());
+        resetData();
+      } else {
+        setSuccess(true);
+        setError("");
+        setInstalling(false);
+      }
+    });
+  }
+
+  const SuccessAlert = () => (
+    <Result
+      status="success"
+      title="Success"
+      subTitle="Successfully sent data to your Lattice. You must confirm all
+  functions on your Lattice for them to be saved.
+  Please confirm or reject the definitions before continuing."
+    />
+  );
+
+  const ErrorAlert = () => (
+    <Result status="error" title="Error" subTitle={error} />
```

Comment:
I think this error screen would look better without a title, which creates a bunch of awkward white space. There's a big red X above it so I don't think the title is needed anyway.

![image](https://user-images.githubusercontent.com/7378490/147492032-e764d78a-9757-4ccc-92bd-9bed5a124f1f.png)